civicrm.module

```diff
@@ -667,10 +667,10 @@ function civicrm_form_data($account, $profile, $reset, $doNotProcess = FALSE) {
     // do not allow edit for anon users in joomla frontend, CRM-4668, unless u have checksum CRM-5228
     $config = CRM_Core_Config::singleton();
     if ($config->userFrameworkFrontend) {
-      CRM_Contact_BAO_Contact_Permission::validateOnlyChecksum($userID, $this);
+      CRM_Contact_BAO_Contact_Permission::validateOnlyChecksum($userID, $edit);
```

Comment:
I had changed `function civicrm_form_data($account, $profile, $reset, $doNotProcess = FALSE)` so that it passes `$edit` to it since it's needed for `validateOnlyChecksum`. My PR had it as `function civicrm_form_data($edit, $account, $profile, $reset, $doNotProcess = FALSE)`. And then passed `$form` to `civicrm_form_data`.